src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  isAlive() {
+    return this.health > 0;
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
The `checkHealth` method removes the animal from the `Animal.alive` array if its health is less than or equal to zero. However, if `checkHealth` is called multiple times, it will attempt to remove the animal from the array again, which could lead to unnecessary operations. Consider adding a condition to check if the animal is already removed from the `Animal.alive` array before attempting to remove it again.